source/scripts/storage/storage.js

```diff
@@ -0,0 +1,145 @@
+// localStorage = {
+//   categories: { // Used to populate explore page (only ids to reduce storage size)
+//       "breakfast": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "lunch": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   savedRecipes: { // Saved recipe page categories (only ids to reduce storage size)
+//       "favorites": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//       "list2": [
+//           "123", // recipe id
+//           "111", // recipe id
+//           "444", // recipe id
+//       ],
+//   },
+//   recipeData: { // Complete storage of available recipes we can query without fetching
+//       "111": "full recipe json object", // these are api results so we never have to modify them
+//       "123": "full recipe json object",
+//       "444": "full recipe json object",
+//       "555": "full recipe json object",
+//       "715": "full recipe json object",
+//       "892": "full recipe json object"
+//   }
+// }
+
+
+// store recipe to local storage
+// add recipeid to category
+// input: category of api fetch, array of recipe json object
+export function storeRecipeData(category, recipeArray) {
+  try {
+    let allData = JSON.parse(localStorage.getItem("recipeData"));
+    if (allData == null) {
+      allData = {};
+    }
+    let currNumElement = Object.keys(allData).length;
+
+    // store recipe
+    let allRecipeId = [];
+    for (let i = 0; i < recipeArray.length; i++) {
+      const uid = currNumElement + i;
+      allRecipeId.push(uid);
+      allData[uid] = recipeArray[i];
+    }
+
+    // add to category list
+    let catData = JSON.parse(localStorage.getItem("categories"));
+    if (catData == null) {
+      catData = {};
+    }
+    if (!(category in catData)) {
+      catData[category] = [];
+    }
+
+    allRecipeId.forEach((rid) => {
+      catData[category].push(rid);
+    });
+
+    localStorage.setItem("categories", JSON.stringify(catData));
+    localStorage.setItem("recipeData", JSON.stringify(allData));
+  } catch (e) {
+    // storage might be full
+    console.log(e);
+  }
+}
+
+// save recipe to a list(add recipeId to savedlist)
+// input: list name to store recipe to, edited recipeObject
+export function saveRecipeToList(listName, recipeId) {
```

Comment:
try to use proper JSDoc style for comments explaining what the whole function does like
```
**
 * 
 * @param {*} listName 
 * @param {*} recipeId 
 */
export function saveRecipeToList(listName, recipeId) {
```

I think if you type /** on the line above the function definition it will autocomplete in vscode. this will help us later when we get to real documentation

Reply:
defintely